src/probnum/kernels/_polynomial.py

```diff
@@ -0,0 +1,67 @@
+"""Polynomial kernel."""
+
+from typing import Optional
+
+import numpy as np
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class Polynomial(Kernel[_InputType]):
+    """Polynomial kernel.
+
+    Covariance function defined by :math:`k(x_0, x_1) = (x_0^\\top x_1 + c)^q`.
```

Comment:
Do the parameters (i.e. `c`) here align with the `Linear` kernel? Would it not be easier to maintain if `Linear` was `Polynomial(exponent=1)`? I am worried about the API if we decide to allow products of kernels, in which case one could get the `Polynomial` by multiplying `Linear` kernels. If `c` does not align with both, we might need some awkward API changes...

Reply:
very good point. I changed the API of `Linear` to match `Polynomial`.